### ISSUE TITLE
Integrate non-line dim. reduction techniques from sklearn

### ISSUE THREAD

**Issue by raamana** [raamana/neuropredict#40]
So far the dimensionality reduction techniques offered are only `feature selection` methods, and feature `transformation` methods.

it's about time to integrate non-linear (and linear) dimensionality reduction techniques, to offer maximum choice (and potential benefit) to users:
https://scikit-learn.org/stable/modules/manifold.html#manifold

**Comment by raamana** [comment]
#44 addresses this.
